File: lib/models/address.dart
class Address {
  String? id;
  String? streetAddress;
  String? city;
  String? state;
  String? country;
  int? zipCode;

  Address({
    this.country,
    this.city,
    this.id,
    this.state,
    this.streetAddress,
    this.zipCode,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['_id'],
      country: json['country'],
      streetAddress: json['streetAddress'],
      city: json['city'],
      state: json['state'],
      zipCode: json['zipCode'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'streetAddress': streetAddress,
      'city': city,
      'country': country,
      'state': state,
      'zipCode': zipCode,
    };
  }
}

File: backend/models/address.ts
import mongoose, { Schema, Document } from "mongoose";

export interface IAddress extends Document {
  streetAddress?: string;
  city?: string;
  country?: string,
  state?: string;
  zipCode?: number;
  user?: Schema.Types.ObjectId,
}

const addressSchema = new Schema<IAddress>(
  {
    user: {
      type: Schema.Types.ObjectId,
      ref: 'users',
      cast: 'Invalid user type'
    },
    streetAddress: {
      type: String,
      cast: "address line datatype is incorrect",
    },
    city: {
      type: String,
      cast: "city datatype is incorrect",
    },
    country: {
      type: String,
      cast: "country datatype is incorrect",
    },
    state: {
      type: String,
      cast: "state datatype is incorrect",
    },
    zipCode: {
      type: Number,
      cast: "zipCode datatype is incorrect",
    },
  },
  { timestamps: false, versionKey: false }
);

const addressModel = mongoose.model('addresses',addressSchema);

export { addressModel, addressSchema };
